app/queues/task_queue.py

```diff
@@ -0,0 +1,50 @@
+import threading
+from dataclasses import dataclass
+from typing import Optional
+
+from persistqueue import SQLiteAckQueue, Empty
+
+from paths import SQLITE_TASKS_PATH
+
+
+@dataclass
+class Task:
+    data_source_id: int
+    function_name: str
+    kwargs: dict
+
+
+@dataclass
+class TaskQueueItem:
+    queue_item_id: int
+    task: Task
+
+
+class TaskQueue(SQLiteAckQueue):
+    __instance = None
+    __lock = threading.Lock()
+
+    @classmethod
+    def get_instance(cls):
+        with cls.__lock:
+            if cls.__instance is None:
+                cls.__instance = cls()
+        return cls.__instance
+
+    def __init__(self):
+        if TaskQueue.__instance is not None:
```

Comment:
- [ ] `__instance is not None` is not a good check whether someone is trying to initialize it (for example someone might try to create a new TaskQueue after the singleton already exists)